### ISSUE TITLE
(opened via other app) Pressing back returns to normal RedReader session instead of returning to other app

### ISSUE THREAD

**Issue by doodhout** [QuantumBadger/RedReader#968]
Given:
* you have another app open
* in that app, there is a reddit link
* you choose to open that reddit link in RedReader

When:
* you are done reading the content of that link in RedReader, you press Android Back button

Then:
* you **expect** to be sent back to the other app from which you opened this reddit link
* but instead you **actually** are returned to whatever you had opened previously in RedReader itself, in its own session so to speak

I hope this use-case, actual behaviour and expected behaviour are explained well enough.

TL;DR: Pressing Back after opening a reddit link from another app should return back to the other app instead of returning to RedReader.
